File: check_vae_keys.rs
use candle_core::Device;
use candle_core::safetensors::MmapedSafetensors;
use std::path::PathBuf;

fn main() -> anyhow::Result<()> {
    let file = PathBuf::from(r"c:\candle-video\models\models--Lightricks--LTX-Video-0.9.5\snapshots\e58e28c39631af4d1468ee57a853764e11c1d37e\vae\diffusion_pytorch_model.safetensors");
    let tensors = MmapedSafetensors::new(file)?;
    let mut keys: Vec<_> = tensors.tensors().iter().map(|(k, _)| k.clone()).collect();
    keys.sort();
    for k in keys {
        if k.contains("latent") || k.contains("mean") || k.contains("std") || k.len() < 20 {
             println!("{}", k);
        }
    }
    Ok(())
}

File: check_vae_keys.py
import torch
from safetensors.torch import load_file
import sys

path = "c:/candle-video/models/models--Lightricks--LTX-Video-0.9.5/vae/diffusion_pytorch_model.safetensors"
try:
    with torch.no_grad():
        sd = load_file(path, device="cpu")
        keys = list(sd.keys())
        print("Keys in VAE safetensors:")
        for k in sorted(keys):
            if "latent" in k:
                print(f"  {k}: {sd[k].shape}")
except Exception as e:
    print(f"Error: {e}")
